program/rust/src/accounts/score.rs

```diff
@@ -0,0 +1,181 @@
+    ) -> Result<(), ProgramError> {
+        let publisher_index = self
+            .publishers
+            .iter()
+            .position(|&x| x == publisher)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == price_account)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        self.set_publisher_permission(publisher_index, symbol_index, false);
+        self.calculate_scores()
+    }
+
+    pub fn add_price(&mut self, symbol: Pubkey) -> Result<(), ProgramError> {
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == symbol)
+            .unwrap_or(self.num_symbols);
+
+        if symbol_index == self.num_symbols {
+            if self.num_symbols == PC_MAX_SYMBOLS as usize {
+                return Err(ProgramError::AccountDataTooSmall);
+            }
+
+            self.symbols[self.num_symbols] = symbol;
+            self.num_symbols += 1;
+        }
+
+        Ok(())
+    }
+
+    pub fn del_price(&mut self, symbol: Pubkey) -> Result<(), ProgramError> {
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == symbol)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        // update symbol list
+        self.symbols[symbol_index] = self.symbols[self.num_symbols - 1];
+        self.symbols[self.num_symbols - 1] = Pubkey::default();
+
+        // update publisher permissions
+        for i in 0..self.num_publishers {
+            let value = self.get_publisher_permission(i, self.num_symbols - 1);
+            self.set_publisher_permission(i, symbol_index, value)
+        }
+
+        self.num_symbols -= 1;
+        self.calculate_scores()
+    }
+
+    pub fn calculate_scores(&mut self) -> Result<(), ProgramError> {
+        let symbol_scores: Vec<u32> = self
+            .symbols
+            .iter()
+            .enumerate()
+            .map(|(j, _)| {
+                let score = self
+                    .publisher_permissions
+                    .iter()
+                    .fold(0, |score, permissions| score + permissions[j] as u32);
+                max(score, self.z)
+            })
+            .collect();
+
+        for i in 0..self.num_publishers {
+            self.scores[i] = self
+                .symbols
+                .iter()
+                .enumerate()
+                .filter(|(j, _)| self.get_publisher_permission(i, *j))
+                .map(|(j, _)| 1f64 / symbol_scores[j] as f64)
+                .sum();
+        }
+        Ok(())
+    }
+}
+
+impl PythAccount for PublisherScoresAccount {
+    const ACCOUNT_TYPE: u32 = PC_ACCTYPE_SCORE;
+    // Calculate the initial size of the account
+    const INITIAL_SIZE: u32 = 1000;
+}
+
+// Unsafe impl because product_list is of size 640 and there's no derived trait for this size
+unsafe impl Pod for PublisherScoresAccount {
```

Comment:
The problem with Pod is the z as u32 adds some padding, so it can't be Pod anymore.
If you make z an u64 then you can use the derive macro

Reply:
I changed them to u64, it simplified a lot of castings as well.